packages/amplication-data-service-generator/src/util/prepare-yaml-file.ts

```diff
@@ -1,11 +1,16 @@
 import YAML from "yaml";
+import { set } from "lodash";
 
 export function prepareYamlFile(
   yamlFileContent: string,
-  updateProperties: { [key: string]: any }
+  updateProperties: { path: string; value: any }[]
 ): string {
   const parsed = YAML.parse(yamlFileContent);
-  const updated = Object.assign(parsed, updateProperties);
+  if (!updateProperties.length) return YAML.stringify(parsed, { nullStr: "~" });
+
+  const updated = updateProperties.map((item) =>
+    set(parsed, item.path, item.value)
+  )[0];
```

Comment:
although set() returns the mutated file - using map() will return an array of files 

I see by the results of the snapshot that somehow it is working - but I still think we should update this code 

you can use forEach or other ways to iterate and then just use the mutated "parsed" object